src/Microsoft.DotNet.Build.Tasks/Targets/Build.Common.targets

```diff
@@ -105,6 +105,17 @@
         DnuRestoreCommand - Used to restore the project packages from project.json
         PackagesDir - Packages are restored to and resolved from this location    
    -->
-  <Import Project="$(MSBuildThisFileDirectory)tests.targets" Condition="'$(IsTestProject)'=='true' and '$(ExcldueTestsImport)'!='true'"/> 
+  <Import Project="$(MSBuildThisFileDirectory)tests.targets" Condition="'$(IsTestProject)'=='true' and '$(ExcldueTestsImport)'!='true'"/>
 
+  <Target Name="CheckDesignTime">
+    <!--
+      Visual Studio does a number of background builds to do a variety of tasks such as resolving references and preparing for intellisense.
+      These are called "design time" builds. You can only determine this state within a target as the properties VS sets are added at build time.
```

Comment:
It isn't actually set for all of the background builds, resolve references in particular. I waffled on terminology here but settled on DesignTime because of the logging variable. I'll add a remark here that $(DesignTimeBuild) isn't always set.